Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.hxx

```diff
@@ -107,9 +112,9 @@ TriangleMeshCurvatureCalculator<TInputMesh>::ComputeGaussCurvature(MeshConstPoin
     CellType *               cellPointer = cellsItr.Value();
     MeshPointIdConstIterator point_ids = cellPointer->GetPointIds();
 
-    MeshPointType v0 = inputMesh->GetPoint(point_ids[0]);
-    MeshPointType v1 = inputMesh->GetPoint(point_ids[1]);
-    MeshPointType v2 = inputMesh->GetPoint(point_ids[2]);
+    v0 = inputMesh->GetPoint(point_ids[0]);
+    v1 = inputMesh->GetPoint(point_ids[1]);
+    v2 = inputMesh->GetPoint(point_ids[2]);
```

Comment:
Did the original code have any warning? As far as I can see, it did already properly initialize v0, v1, and v2! So then, why should it be changed, in your opinion?

Reply:
When I build original code with Python Wrapping I get following warnings:

```
Linking CXX shared module Wrapping/Generators/Python/itk/_ITKMeshPython.so
../Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.hxx: In member function ‘Compute’:
../Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.hxx:127:11: warning: ‘MEM[(ValueType &)&v1 + 8]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  127 |     e0[2] -= v0[2];
      |           ^
../Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.hxx:88:21: note: ‘MEM[(ValueType &)&v1 + 8]’ was declared here
   88 |   MeshPointType v0, v1, v2;
      |                     ^
../Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.hxx:127:11: warning: ‘MEM[(ValueType &)&v0 + 8]’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  127 |     e0[2] -= v0[2];
      |           ^
../Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.hxx:88:17: note: ‘MEM[(ValueType &)&v0 + 8]’ was declared here
   88 |   MeshPointType v0, v1, v2;
   ```

However, no such warning is thrown when only building C++ code. The GetPoint method definition from PointSet class is shown below:

```
PointType point;
bool exist = m_PointsContainer->GetElementIfIndexExists(ptId, &point);
if (!exist)
{
    itkExceptionMacro("Point id doesn't exist: " << ptId);
}
return point;
```

So it takes care of the cases when the id does not exists by throwing exception but I don't know how this behaves with Python Wrapping.

Reply:
@PranjalSahu  Interesting, thanks for your explanation! I _think_ the warning is a false alarm (although of course, we do not want to have warnings anyway).  I _think_  the compiler suspects that, because `GetElementIfIndexExists(ptId, &point)` might not assign anything to `point`,  the `return` statement might be executed while `point` is still uninitialized. Right?

Reply:
Will test and reply here.

Reply:
I have tested the change you suggested and it didn't work. And the change that I did also does not work. 

I also did one more test. At line 194 in [GetPoint(PointIdentifier ptId)](https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3rc02/Modules/Core/Common/include/itkPointSet.hxx#L194) there is another exception for absence of m_PointsContainer. I commented all itkExceptionMacro and the warning went away. The problem is occurring wherever the return is itkExceptionMacro, which probably translates to null in Python. I believe it will need a major change to handle such warnings while doing Python wrapping.

Reply:
@PranjalSahu Hmmm, weird!

> The problem is occurring wherever the return is itkExceptionMacro, which probably translates to null in Python

I don't know that much about the Python wrapping, but a call to itkExceptionMacro is unconditionally translated to something like:
```
  {
    std::ostringstream exceptionDescriptionOutputStringStream; 
    exceptionDescriptionOutputStringStream << "ITK ERROR: " x;
    throw ::itk::ExceptionObject(
      std::string{ __FILE__ }, __LINE__, exceptionDescriptionOutputStringStream.str(), std::string{ITK_LOCATION });
  }
```

https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3rc02/Modules/Core/Common/include/itkMacro.h#L525-L541

So really I have to admit I don't entirely understand what's going on!

Could you possibly try the following as well, as the last few lines of `PointSet::GetPoint(PointIdentifier ptId)`, instead of the adjustment I previously suggested? (Directly after the comment which says "Ask the container if the point identifier exists.")

```
  if (!m_PointsContainer->IndexExists(ptId))
  {
    itkExceptionMacro("Point id doesn't exist: " << ptId);
  }
  return (*m_PointsContainer)[static_cast<std::size_t>(ptId)];
```

You never know, maybe the compiler gets happy when there is an unconditional return statement at the end!

Reply:
ok will test and reply here.

Reply:
No this also does not work.